test/gtest/ucp/test_ucp_ep_reconfig.cc

```diff
@@ -288,6 +298,28 @@ test_ucp_ep_reconfig::entity::get_address(const ucp_tl_bitmap_t &tl_bitmap) cons
     return address;
 }
 
+void test_ucp_ep_reconfig::init_buffers(mem_buffer_vec_t &sbufs,
+                                        mem_buffer_vec_t &rbufs, size_t size,
+                                        bool bidirectional)
+{
+    auto num_bufs = bidirectional ? 2 * num_iterations : num_iterations;
+
+    for (unsigned i = 0; i < num_bufs; ++i) {
+        sbufs.push_back(
+                mem_buffer_p(new mem_buffer(size, UCS_MEMORY_TYPE_HOST, i)));
+        rbufs.push_back(mem_buffer_p(
+                new mem_buffer(size, UCS_MEMORY_TYPE_HOST, ucs::rand())));
+    }
+}
+
+void test_ucp_ep_reconfig::pattern_check(const mem_buffer_vec_t &rbufs,
+                                         size_t size, bool bidirectional) const
+{
+    for (unsigned i = 0; i < rbufs.size(); ++i) {

```

Comment:
for (auto rbuf: rbufs) { .. }

Reply:
this will require extra counter to replace "i"